server/block/water.go

```diff
@@ -112,6 +113,37 @@ func (w Water) ScheduledTick(pos cube.Pos, wo *world.World, _ *rand.Rand) {
 	tickLiquid(w, pos, wo)
 }
 
+// RandomTick ...
+func (w Water) RandomTick(pos cube.Pos, wo *world.World, _ *rand.Rand) {
+	if w.Depth != 8 {
+		return
+	}
+	if wo.Temperature(pos) > 0.0 {
+		return
+	}
+	if wo.Light(pos) > 11 {
```

Comment:
```suggestion
	if wo.Light(pos) > 12 {
```

From the Minecraft wiki:
> **[Water](https://minecraft.fandom.com/wiki/Water) source blocks in a snowy [biome](https://minecraft.fandom.com/wiki/Biome) eventually freeze into ice if exposed to the sky from directly above, the [light level](https://minecraft.fandom.com/wiki/Light_Level) immediately adjacent to the water block on all sides is less than 13**, and there is at least one horizontally adjacent non-water (and non-[waterlogged](https://minecraft.fandom.com/wiki/Waterlogging)) block. This can happen at any [time of day](https://minecraft.fandom.com/wiki/Daylight_cycle), and in any [weather](https://minecraft.fandom.com/wiki/Weather) condition. **If the highest adjacent light level is 12, an ice block alternately melts and re-freezes when it receives a [block tick](https://minecraft.fandom.com/wiki/Tick#Random_tick).**